### PR TITLE
tls: Add subjectLocalCertificate to get local cert subject.

### PR DESCRIPTION
**Pull request by saumoh**
_Description:_
Add subjectLocalCertificate().
In filling out the external auth proto's [Principal](https://github.com/envoyproxy/data-plane-api/blob/aa73175ddf5e229dd4d97a019e5312ba1263d252/api/auth/external_auth.proto#L60) It may be required that the local certificates subject needs to be used.
Here I am adding the missing api that in tls/ssl so that the value may be extracted from the certificate.

See also https://github.com/envoyproxy/envoy/issues/2291

_Risk Level_: Low

_Testing_:
Wrote ut that test the subjects in ssl (local and peer). Also added test to check for local SAN.

_Docs Changes:_
N/A
_Release Notes_:
N/A

### CODE REVIEW

**Comment by mattklein123** [comment]
@saumoh please merge master and fix tests also.

File: include/envoy/ssl/connection.h [hunk 1]
```diff
@@ -25,6 +25,12 @@ class Connection {
    **/
   virtual std::string uriSanLocalCertificate() PURE;
 
+  /**
+   * @return the subject field of the local certificate in RFC 2253 format. Returns "" if there is
```
Comment (ggreenway) [inline include/envoy/ssl/connection.h]: +1 on citing the RFC in the comment.  Can you please also put a simple example value, so I don't have to look up the RFC to know approximately what this function returns?

**Comment by saumoh** [comment]
@ggreenway @mattklein123 Thx for the review. i've addressed review feedback and tests are passing now :-)

File: include/envoy/ssl/connection.h (hunk 1 above)
Comment (ggreenway) [reply include/envoy/ssl/connection.h]: @saumoh Can you either make the doc change here or comment on why it's not a good idea?

Comment (saumoh) [reply include/envoy/ssl/connection.h]: @ggreenway I should have mentioned that i added some sample values in one of the follow up  commits.
https://github.com/envoyproxy/envoy/pull/2319/commits/409f159227bbf6d78516afec480914cb32d148f2#diff-ded6386c66f63a1375432303053ca7eeR290

**Pull request merged**